packages/glimmer-runtime/lib/compiled/expressions/concat.ts

```diff
@@ -31,10 +31,28 @@ class ConcatReference extends CachedReference<string> {
   }
 
   protected compute(): string {
-    let parts = new Array<string>(this.parts.length);
+    let parts = new Array<string>();
```

Comment:
Maybe this array could have fixed size? I think I remember reading that making arrays grow deoptimizes.